### PR TITLE
Used string interpolation in notes_controller

### LINKED ISSUE #11446: Rubocop - Use string interpolation `(notes_controller)`
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from underrepresented groups in free and open-source software!

We know that the process of creating a pull request is one of the biggest barriers for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new** and looking through our general [help wanted](https://github.com/publiclab/plots2/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Claim the issue by commenting. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!
<details>
  <summary>💡 <i>Learn how to claim</i> 🙋</summary>
  <h3>Claiming an issue</h3>
  <p>Unless the issue is marked as reserved for someone, you can just say "I'd like to try this!" and then you've claimed it - no need to wait for someone to assign it to you. Just be sure you link your pull request (PR) to this issue so we can see where your solution is.</p>
  <p>And open one early if possible - even before you've completed it with additional commits - and others can help you figure out any issues you may face.</p>
</details>

- [ ] 📝 **Update** the file [app/controllers/notes_controller.rb](https://github.com/publiclab/plots2/blob/main/app/controllers/notes_controller.rb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -140,7 +140,7 @@ class NotesController < ApplicationController
       return
     elsif params[:draft] == "true"
       token = SecureRandom.urlsafe_base64(16, false)
-      @node.slug = @node.slug + " token:" + token
+      @node.slug = "#{@node.slug} token:#{token}"
       @node.save!
     end

@@ -153,7 +153,7 @@ class NotesController < ApplicationController
       if params[:event] == 'on'
         @node.add_tag('event', current_user)
         @node.add_tag('event:rsvp', current_user)
-        @node.add_tag('date:' + params[:date], current_user) if params[:date]
+        @node.add_tag("date:#{params[:date]}", current_user) if params[:date]
       end

       @node.add_tag('first-time-poster', current_user) if current_user.first_time_poster
@@ -273,7 +273,7 @@ class NotesController < ApplicationController
         if request.xhr?
           render plain: "#{@node.path(format)}?_=#{Time.now.to_i}"
         else
-          redirect_to URI.parse(@node.path(format)).path + '?_=' + Time.now.to_i.to_s
+          redirect_to "#{URI.parse(@node.path(format)).path}?_=#{Time.now.to_i}"
         end
       else
         flash[:error] = I18n.t('notes_controller.edit_not_saved')
@@ -301,13 +301,13 @@ class NotesController < ApplicationController
               render plain: I18n.t('notes_controller.content_deleted')
             else
               flash[:notice] = I18n.t('notes_controller.content_deleted')
-              redirect_to '/dashboard' + '?_=' + Time.now.to_i.to_s
+              redirect_to "/dashboard?_=#{Time.now.to_i}"
             end
           end
         end
       else
         flash[:error] = I18n.t('notes_controller.more_than_one_contributor')
-        redirect_to '/dashboard' + '?_=' + Time.now.to_i.to_s
+        redirect_to "/dashboard?_=#{Time.now.to_i}"
     end
     else
       prompt_login
@@ -328,7 +328,7 @@ class NotesController < ApplicationController
   def author_topic
     @user = User.find_by(name: params[:author])
     @tagnames = params[:topic].split('+')
-    @title = @user.name + " on '" + @tagnames.join(', ') + "'"
+    @title = "#{@user.name} on '#{@tagnames.join(', ')}'"
     @notes = @user.notes_for_tags(@tagnames)
     @unpaginated = true
     render template: 'notes/index'
@@ -391,8 +391,8 @@ class NotesController < ApplicationController
     # leave a comment
     @comment = @node.add_comment(subject: 'rsvp', uid: current_user.uid, body: 'I will be attending!')
     # make a tag
-    @node.add_tag('rsvp:' + current_user.username, current_user)
-    redirect_to URI.parse(@node.path).path + '#comments'
+    @node.add_tag("rsvp:#{current_user.username}", current_user)
+    redirect_to "#{URI.parse(node.path).path}#comments"
   end

   # Updates title of a wiki page, takes id and title as query string params. maps to '/node/update/title'
@@ -400,21 +400,21 @@ class NotesController < ApplicationController
     node = Node.find params[:id].to_i
     unless current_user && current_user == node.author
       flash.keep[:error] = I18n.t('notes_controller.author_can_edit_note')
-      return redirect_to URI.parse(node.path).path + "#comments"
+      return redirect_to "#{URI.parse(node.path).path}#comments"
     end
     node.update(title: params[:title])
-    redirect_to URI.parse(node.path).path + "#comments"
+    redirect_to "#{URI.parse(node.path).path}#comments"
   end

   def publish_draft
     @node = Node.find(params[:id])
-    if current_user && current_user.uid == @node.uid || current_user.can_moderate? || @node.has_tag("with:#{current_user.username}")
+    if (current_user && current_user.uid == @node.uid) || current_user.can_moderate? || @node.has_tag("with:#{current_user.username}")
       @node.path = @node.generate_path
       @node.slug = @node.slug.split('token').first
       @node['created'] = DateTime.now.to_i # odd assignment needed due to legacy Drupal column types
       @node['changed'] = DateTime.now.to_i
       revision = @node.latest
-      revision['timestamp'] = DateTime.now.to_i # odd assignment needed due to legacy Drupal column types
+      revision['timestamp'] = DateTime.now.to_i # odd assignment needed due to legacy Drupal column types
       revision.save
       @node.save
       @node.publish
@@ -467,11 +467,13 @@ class NotesController < ApplicationController
   end

   def new_preview_note
-    Node.new_preview_note(uid: current_user.uid,
+    Node.new_preview_note(
+      uid: current_user.uid,
       title: params[:title],
       body: params[:body],
       main_image: params[:main_image],
-      location: params[:location])
+      location: params[:location]
+    )
   end

   def not_draft_and_user_is_first_time_poster?
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there.

- [ ] 🏁 **Done** Ask in the comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming the `first-timers-only` issue. We're happy to keep it assigned to you as long as you need it if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance!

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below, and [reach out in our chatroom](https://publiclab.org/chat)!

### PR DESCRIPTION
**Pull request by IdayatSanni**
Fixes #11446 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by IdayatSanni** [comment]
@TildaDares hi, I just created this request but it is saying I have some issues. how do I fix it?

File: app/controllers/notes_controller.rb [hunk 1]
```diff
@@ -470,11 +470,13 @@ def new_note
...
   def new_preview_note
-    Node.new_preview_note(uid: current_user.uid,
+    Node.new_preview_note(
++      uid: current_user.uid,
       title: params[:title],
       body: params[:body],
       main_image: params[:main_image],
-      location: params[:location])
+      location: params[:location]
++    )
```
Comment (TildaDares) [inline app/controllers/notes_controller.rb:479]:
```suggestion
    )
```

File: app/controllers/notes_controller.rb [hunk 2]
```diff
@@ -470,11 +470,13 @@ def new_note
   end
 
   def new_preview_note
-    Node.new_preview_note(uid: current_user.uid,
+    Node.new_preview_note(
++      uid: current_user.uid,
```
Comment (TildaDares) [inline app/controllers/notes_controller.rb:474]:
```suggestion
      uid: current_user.uid,
```

File: app/controllers/notes_controller.rb [hunk 3]
```diff
@@ -273,7 +273,7 @@ def update
         if request.xhr?
           render plain: "#{@node.path(format)}?_=#{Time.now.to_i}"
         else
-          redirect_to URI.parse(@node.path(format)).path + '?_=' + Time.now.to_i.to_s
+          redirect_to "#{URI.parse(@node.path(format)).path}?_=#{Time.now.to_i}
```
Comment (TildaDares) [inline app/controllers/notes_controller.rb:276]:
```suggestion
          redirect_to "#{URI.parse(@node.path(format)).path}?_=#{Time.now.to_i}"
```

**Review by TildaDares — CHANGES_REQUESTED** [review]
@IdayatSanni There was a missing quotation mark.

**Comment by IdayatSanni** [comment]
Hello, thank you for the review. I will work on it

**Comment by IdayatSanni** [comment]
Hello @TildaDares I have made corrections but it is still showing issues. What exactly am I doing wrong please?

File: app/controllers/notes_controller.rb (hunk 1 above)
Comment (TildaDares) [reply app/controllers/notes_controller.rb]: Hi @IdayatSanni. This has not been resolved.

File: app/controllers/notes_controller.rb (hunk 2 above)
Comment (TildaDares) [reply app/controllers/notes_controller.rb]: Same with this.

Comment (IdayatSanni) [reply app/controllers/notes_controller.rb]: Hello, am I understanding that the problem here is that I have to delete one space and bring it forward?

Comment (TildaDares) [reply app/controllers/notes_controller.rb]: @IdayatSanni The issue is that you added the plus sign in the code.

**Review by TildaDares — APPROVED** [review]
Amazing work @IdayatSanni. Thank you!!

**Comment by TildaDares** [comment]
Congratulations on merging your PR @IdayatSanni 🎉

**Comment by IdayatSanni** [comment]
Thank you @TildaDares! I appreciate the guidance.

**Pull request merged**